ci/travis/ci.sh

```diff
@@ -127,24 +313,90 @@ build() {
+  case "${OSTYPE}" in
+    linux*)
+      if [ "${LINUX_WHEELS-}" = 1 ]; then
+        # Mount bazel cache dir to the docker container.
+        # For the linux wheel build, we use a shared cache between all
+        # wheels, but not between different travis runs, because that
+        # caused timeouts in the past. See the "cache: false" line below.
+        local MOUNT_BAZEL_CACHE=(-v "${HOME}/ray-bazel-cache":/root/ray-bazel-cache -e TRAVIS=true -e TRAVIS_PULL_REQUEST="${TRAVIS_PULL_REQUEST:-false}" -e encrypted_1c30b31fe1ee_key="${encrypted_1c30b31fe1ee_key-}" -e encrypted_1c30b31fe1ee_iv="${encrypted_1c30b31fe1ee_iv-}")
+
+        # This command should be kept in sync with ray/python/README-building-wheels.md,
+        # except the "${MOUNT_BAZEL_CACHE[@]}" part.
+        suppress_output docker run --rm -w /ray -v "${PWD}":/ray "${MOUNT_BAZEL_CACHE[@]}" -e TRAVIS_COMMIT="${TRAVIS_COMMIT}" rayproject/arrow_linux_x86_64_base:python-3.8.0 /ray/python/build-wheel-manylinux1.sh
+      fi
+      ;;
+    darwin*)      
+      if [ "${MAC_WHEELS-}" = 1 ]; then
+        # This command should be kept in sync with ray/python/README-building-wheels.md.
+        suppress_output "${WORKSPACE_DIR}"/python/build-wheel-macos.sh
+      fi
+      ;;
+    msys*)
+      if [ "${WINDOWS_WHEELS-}" = 1 ]; then
+        (
+          cd "${WORKSPACE_DIR}"/python
+          python setup.py --quiet bdist_wheel
+        )
+      fi
+      ;;
+  esac
+}
+
+run() {
+  local result=0 flush_logs=0
+  ##### BEGIN TASKS #####
+
+  if [ "${LINT-}" = 1 ]; then
+    lint
+  fi
+
+  if [ "${RAY_DEFAULT_BUILD-}" = 1 ]; then
+    test_python
+  fi
+
+  if [ "${RAY_INSTALL_JAVA-}" = 1 ]; then
+    "${WORKSPACE_DIR}"/java/test.sh
+  fi
 
-    # This command should be kept in sync with ray/python/README-building-wheels.md,
-    # except the "${MOUNT_BAZEL_CACHE[@]}" part.
-    suppress_output docker run --rm -w /ray -v "${PWD}":/ray "${MOUNT_BAZEL_CACHE[@]}" -e TRAVIS_COMMIT="${TRAVIS_COMMIT}" -ti rayproject/arrow_linux_x86_64_base:python-3.8.0 /ray/python/build-wheel-manylinux1.sh
+  if [ "${TESTSUITE-}" = cpp_worker ]; then
+    bazel test --config=ci //cpp:all --build_tests_only --test_output=streamed
   fi
 
-  if [ "${MAC_WHEELS-}" = 1 ]; then
-    # This command should be kept in sync with ray/python/README-building-wheels.md.
-    suppress_output "${WORKSPACE_DIR}"/python/build-wheel-macos.sh
+  local should_test_wheels=""
+  case "${OSTYPE}" in
+    linux*) should_test_wheels="${LINUX_WHEELS-}";;
+    darwin*) should_test_wheels="${MAC_WHEELS-}";;
+  esac
+  if [ "${should_test_wheels}" = 1 ]; then
+    "${WORKSPACE_DIR}"/ci/travis/test-wheels.sh || { result=$? && flush_logs=1; }
+  fi
+
+  ##### END TASKS #####
```

Comment:
what is this? We should get rid of it.

Reply:
You mean the comment? It's just indicating that all the build tasks need to go before it. Because the step after needs to come last (flushing the CI logs), otherwise someone might add stuff after that by accident?

Reply:
No, I mean the log flushing from below!

Reply:
I didn't add that, it used to be there. I just moved it to this file. But I'm pretty sure it's necessary.